lib/functions.php

```diff
@@ -785,7 +812,11 @@ function __coroutineAdvance(CoroutineState $cs) {
                 $result = (PHP_MAJOR_VERSION >= 7) ? $cs->generator->getReturn() : null;
                 $cs->promisor->succeed($result);
             }
-        } elseif ($yielded instanceof Promise) {
+        } elseif ($yielded instanceof Promise || $yielded instanceof ReactPromise) {
+            if ($yielded instanceof ReactPromise) {
+                $yielded = __transformReactPromise($yielded);
+            }
+
```

Comment:
For speed, it's probably better to move it further down and just duplicate the `if` block content.